File: liqp-core/src/main/java/liqp/filters/Escape.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Escape extends Filter {

    /*
     * escape(input)
     *
     * escape a string
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        String str = super.asString(value);

        return str.replace("&", "&amp;")
                .replace("<", "&lt;")
                .replace(">", "&gt;")
                .replace("\"", "&quot;");
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Escape.kt
package liqp.filter

import liqp.context.LContext

class Escape : LFilter() {

  /**
   * escape(input)
   *
   * escape a string
   */
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    val str = context.asString(value)
    return str?.replace("&", "&amp;")
        ?.replace("<", "&lt;")
        ?.replace(">", "&gt;")
        ?.replace("\"", "&quot;")
  }
}
